looper/cli_pydantic.py

```diff
@@ -0,0 +1,143 @@
+"""
+CLI script using `pydantic-argparse` for parsing of arguments
+
+Arguments / commands are defined in `command_models/` and are given, eventually, as
+`pydantic` models, allowing for type-checking and validation of arguments.
+
+Note: this is only a test script so far, and coexists next to the current CLI
+(`cli_looper.py`), which uses `argparse` directly. The goal is to eventually
+replace the current CLI with a CLI based on above-mentioned `pydantic` models,
+but whether this will happen with `pydantic-argparse` or another, possibly self-
+written library is not yet clear.
+It is well possible that this script will be removed again.
+"""
+
+import os
+import sys
+
+import pydantic_argparse
+import yaml
+from pephubclient import PEPHubClient
+
+from divvy import select_divvy_config
+
+from . import __version__
+from .cli_looper import _proc_resources_spec
+from .command_models.commands import TopLevelParser
+from .const import *
+from .divvy import DEFAULT_COMPUTE_RESOURCES_NAME, select_divvy_config
+from .exceptions import *
+from .looper import *
+from .parser_types import *
+from .project import Project, ProjectContext
+from .utils import (
+    dotfile_path,
+    enrich_args_via_cfg,
+    is_registry_path,
+    read_looper_dotfile,
+)
+
+
+def main() -> None:
+    parser = pydantic_argparse.ArgumentParser(
+        model=TopLevelParser,
+        prog="looper",
+        description="pydantic-argparse demo",
+        add_help=True,
+    )
+    args = parser.parse_typed_args()
+    print(args)
+    print("#########################################")
+
+    # here comes adapted `cli_looper.py` code
+    looper_cfg_path = os.path.relpath(dotfile_path(), start=os.curdir)
+    try:
+        looper_config_dict = read_looper_dotfile()
+
+        for looper_config_key, looper_config_item in looper_config_dict.items():
+            print(looper_config_key, looper_config_item)
+            setattr(args, looper_config_key, looper_config_item)
+
+    except OSError:
+        parser.print_help(sys.stderr)
+        raise ValueError(
+            f"Looper config file does not exist. Use looper init to create one at {looper_cfg_path}."
+        )
+
+    args = enrich_args_via_cfg(args, parser, False)
+    divcfg = (
+        select_divvy_config(filepath=args.run.divvy)
+        if hasattr(args.run, "divvy")
+        else None
+    )
+    # Ignore flags if user is selecting or excluding on flags:
+    if args.sel_flag or args.exc_flag:
+        args.ignore_flags = True
+
+    # Initialize project
+    if is_registry_path(args.config_file):
+        if vars(args)[SAMPLE_PL_ARG]:
+            p = Project(
+                amendments=args.amend,
+                divcfg_path=divcfg,
+                runp=args.command == "runp",
+                project_dict=PEPHubClient()._load_raw_pep(
+                    registry_path=args.config_file
+                ),
```

Comment:
If `looper` can call this protected method on a `PEPHubClient`, is it worth perhaps unprotecting that method?

Reply:
Yeah, this is good point! I am currently working on pephubclient, and will change it in the future release